cmake/compiler/gcc/target_baremetal.cmake

```diff
@@ -5,6 +5,8 @@ macro(toolchain_cc_nostdinc)
     NOT COMPILER STREQUAL "xcc" AND
     NOT CONFIG_NATIVE_APPLICATION)
     zephyr_compile_options( -nostdinc)
+  else()
+    set(NOSTDINC "")
```

Comment:
It's not OK to mutate this variable. What is OK, is not using it, or not populating it in the first place.

This variable can under no circumstances be mutated here.